integrations/box/syncs/files.ts

```diff
@@ -0,0 +1,103 @@
+import type { BoxDocument, BoxMetadata, NangoSync, ProxyConfiguration } from '../../models';
+import type { BoxFile, ListFolderItemsResponse } from '../types.js';
+
+export default async function fetchData(nango: NangoSync) {
+    const metadata = await nango.getMetadata<BoxMetadata>();
+    const files = metadata?.files ?? [];
+    const folders = metadata?.folders ?? [];
+    const batchSize = 100;
+
+    if (files.length === 0 && folders.length === 0) {
+        throw new Error('Metadata for files or folders is required.');
+    }
+
+    for (const folder of folders) {
+        await fetchFolder(nango, folder);
+    }
+
+    let batch: BoxDocument[] = [];
+    for (const file of files) {
+        const metadata = await getFileMetadata(nango, file);
+        batch.push({
+            id: metadata.id,
+            name: metadata.name,
+            modified_at: metadata.modified_at,
+            download_url: metadata.shared_link?.download_url,
+            field_id: metadata.id
```

Comment:
what is the `field_id` used for / why is this included?

Reply:
this is to help the user understand that that the field_id in the api docs is the same as the id field